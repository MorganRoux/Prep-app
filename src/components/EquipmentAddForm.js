import React from 'react'
import { Paper, TableBody, TableRow, TableCell, TableFooter } from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';
import Icon from '@material-ui/core/Icon';

import { addEquipment } from '../actions/equipment'
import { connect } from 'react-redux'
import { withSnackbar } from 'notistack'

export class EquipmentAddForm extends React.Component {
    
    constructor(props) {
        super(props);

        this.state = {
            quantity: '',
            publicName: ''
        }
    }

    handleCloseSnackBar = (event, reason) => {
        if (reason === 'clickaway') {
          return;
        }
    
        this.setState({ snackBarOpen: false });
      };

    onQuantityChange = (e) => {
        const quantity = e.target.value;
        if(quantity.match(/^\d*$/)) {
            this.setState( () => ({quantity}));
        }
    }

    onPublicNameChange = (e) => {
        const publicName = e.target.value;
        this.setState( () => ({publicName}));
    }

    onAdd = () => {

        // test the inputs
        if(!this.state.publicName || !this.state.quantity) {
            this.props.enqueueSnackbar('Informations erronnées', {variant:'error'});
            return;
        }

        // search the item in the stocklist
        const itemToAdd = this.props.stocklist.find( (item) => {
            return (item.publicName === this.state.publicName)
        });
        if (!itemToAdd) { 
            this.props.enqueueSnackbar('Élement introuvable', {variant:'error'});
            return;
        }

        // add item
        this.props.addEquipment({
            ...itemToAdd,
            quantity: this.state.quantity
        });
        const s = this.state.quantity === '1' ? '':'s'
        this.props.enqueueSnackbar(
            `${this.state.quantity}x ${this.state.publicName} ajouté${s}`, {variant:'success'});

        this.setState( () => ({
            quantity: '',
            publicName:''
        }));
    }

    render () {
        return (
            <TableFooter>
                <TableRow className = "addForm-row">
                    <TableCell></TableCell>
                    <TableCell className= "addForm-cell-1">
                        <input 
                        type="text"
                        name= "quantity"
                        autoFocus
                        placeholder=""
                        value={this.state.quantity}
                        onChange = {this.onQuantityChange}
                        />
                    </TableCell>
                    <TableCell className= "addForm-cell-2">
                        <input 
                        type="text"
                        name= "publicName"
                        placeholder=""
                        value={this.state.publicName}
                        onChange = {this.onPublicNameChange}
                        />
                    </TableCell>
                    <TableCell className= "addForm-cell-3"></TableCell>
                    <TableCell className= "addForm-cell-4">
                        <IconButton className="add-button" aria-label="Add" onClick = {this.onAdd}>
                            <Icon color="primary">
                            add_circle
                            </Icon>
                        </IconButton>
                    </TableCell>
                </TableRow>
                </TableFooter>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        equipements: state.equipements,
        stocklist: state.stocklist,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        addEquipment: (item) => dispatch(addEquipment(item))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(withSnackbar(EquipmentAddForm))