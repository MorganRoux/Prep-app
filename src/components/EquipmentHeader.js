import React from 'react'
import { connect } from 'react-redux';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import { setTextFilter } from '../actions/filters';
import { startLogout } from '../actions/user';

export class EquipmentHeader extends React.Component {

    constructor (props) {
        super(props);
        this.state = {
            searchText : ''
        }
    }

    onSearchChange = (e) => {

        const searchText = e.target.value;
        this.setState(() => ({searchText}));
        this.props.setTextFilter(searchText);
    }
    
    render () {
        return (
            <div style = {{display: 'flex', justifyContent: 'space-between'}}>
                <div style = {{display: 'flex', justifyContent: 'flex-start'}}>
                    <IconButton>
                        <MenuIcon />
                    </IconButton>
                    <h1>Equipment List</h1>
                </div>
                <div style = {{display: 'flex', justifyContent: 'flex-end'}}>
                    <InputBase 
                        className="search-field"
                        placeholder="Rechercher" 
                        value={this.state.searchText}
                        onChange={this.onSearchChange}
                    />
                    <IconButton>
                        <SearchIcon />
                    </IconButton>
                    <button className = "logout-button" onClick={this.props.startLogout}>Logout</button>
                </div>
            </div>
    )};

}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter : (text) =>  dispatch(setTextFilter(text)),
    startLogout : () => dispatch(startLogout())
});

export default connect(null, mapDispatchToProps)(EquipmentHeader)