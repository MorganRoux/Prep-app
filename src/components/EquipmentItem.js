import React from 'react'
import { connect } from 'react-redux';
import {TableBody, TableRow, TableCell } from '@material-ui/core';
import { removeEquipment } from '../actions/equipment';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import Checkbox from '@material-ui/core/Checkbox';
import { withSnackbar } from 'notistack';
import { SortableElement } from 'react-sortable-hoc';

export class EquipmentItem extends React.Component {
    constructor(props) {
        super(props);
        const stockItem = props.stocklist.find((item) => {
            return ( item.stockName === props.equipment.stockName )
        });
        this.state = {
            stockName: props.equipment.stockName,
            id: props.equipment.id,
            quantity: props.equipment.quantity,
            stockItem: stockItem
        }
        
    }
    
    onRemove = () => {
        this.props.removeEquipment(this.state.id);
        this.props.enqueueSnackbar('Item supprimÃ©', {variant:'success'});
    }
    
    onQuantityChange = () => {

    }

    onPublicNameChange = () => {
        console.log('change !');
    }

    onStockNameChange = () => {

    }
    
    render() {
        return (
            <TableRow key={`${this.state.id}-row`}>
                <TableCell><Checkbox /></TableCell>
                <TableCell key = {`${this.state.id}-cell-1`}>{this.state.quantity}</TableCell>
                <TableCell key = {`${this.state.id}-cell-2`}>{this.state.stockItem.publicName}</TableCell>
                <TableCell key = {`${this.state.id}-cell-3`}>{this.state.stockItem.stockName}</TableCell>
                <TableCell key = {`${this.state.id}-cell-4`}>
                    <IconButton 
                        className = {'delbutton'} 
                        aria-label="Delete" 
                        onClick = {this.onRemove}>
                    <DeleteIcon />
                    </IconButton>
                </TableCell>   
            </TableRow>
        );
    };
}

const mapStateToProps = (state, props) => {
    return {
        equipments: state.equipments,
        filters: state.filters,
        stocklist : state.stocklist
    };
} 

const mapDispatchToProps = (dispatch, props) => {
    return {
        removeEquipment: (id) => dispatch(removeEquipment(id)),
        editEquipment: () => dispatch(editEquipment()),
        addEquipment: () => dispatch(addEquipment())
    };
}

const EquipmentItemWithBody = (props) => (
    <TableBody>
        <EquipmentItem {...props} />
    </TableBody>
);

// export default connect(mapStateToProps,mapDispatchToProps)(withSnackbar(EquipmentItem));

export const EquipmentKitItem = connect(mapStateToProps,mapDispatchToProps)(withSnackbar(EquipmentItem));
export const EquipmentListItemSortable = connect(mapStateToProps,mapDispatchToProps)(withSnackbar(SortableElement(EquipmentItemWithBody)));
