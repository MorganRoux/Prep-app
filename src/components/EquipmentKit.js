import React from 'react';
import { connect } from 'react-redux';
import { removeEquipment } from '../actions/equipment';
import IconButton from '@material-ui/core/IconButton';
import Icon from '@material-ui/core/Icon';
import DeleteIcon from '@material-ui/icons/Delete';
import {TableBody, TableRow, TableCell } from '@material-ui/core';
import Checkbox from '@material-ui/core/Checkbox';
import { withSnackbar } from 'notistack';

export class EquipmentKit extends React.Component{
    constructor(props) {
        super(props);

        this.state = {
            ...props.kit,
            expand: true
        }
    }
    onRemove = () => {
        this.props.removeEquipment(this.state.id);
        this.props.enqueueSnackbar('Kit supprimé', {variant:'success'});
    }

    onExpand = () => {
        this.setState( (prevState) => ({
            expand: !prevState.expand
        }));
    }

    render() {
        return (
            <TableBody key = {`${this.state.id}-body`}>
                <TableRow key = {`${this.state.id}-row`}>
                    <TableCell key = {`${this.state.id}-cell-1`}>
                        <Checkbox />
                        <IconButton className="expand-button" aria-label="Expand" onClick = {this.onExpand}>
                            <Icon color="primary">
                            {this.state.expand ? 'arrow_drop_down' : 'arrow_right'}
                            </Icon>
                        </IconButton>
                    </TableCell>
                    <TableCell key = {`${this.state.id}-cell-2`}>{this.state.quantity}</TableCell>
                    <TableCell key = {`${this.state.id}-cell-3`} colSpan="2">EquipmentKit</TableCell>
                    <TableCell key = {`${this.state.id}-cell-4`}>
                        <IconButton className="delbutton" aria-label="Delete" onClick = {this.onRemove}>
                        <DeleteIcon />
                        </IconButton>
                    </TableCell>
                </TableRow>
                { this.state.expand && this.props.children }
            </TableBody>
        );
    }
}

const mapStateToProps = (state) => {
    return {
         equipments: state.equipments
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        removeEquipment: (key) => dispatch(removeEquipment(key))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(withSnackbar(EquipmentKit))