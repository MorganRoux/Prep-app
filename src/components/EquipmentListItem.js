import React from 'react'
import { connect } from 'react-redux';
import Table from '@material-ui/core/Table';
import { Paper, TableBody, TableRow, TableCell } from '@material-ui/core';
import { removeEquipment } from '../actions/equipment';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import Checkbox from '@material-ui/core/Checkbox';

export class EquipmentListItem extends React.Component {
    constructor(props) {
        super(props);
        const stockItem = props.stocklist.find((item) => {
            return ( item.stockName === props.equipment.stockName )
        });
        this.state = {
            stockName: props.equipment.stockName,
            id: props.equipment.id,
            quantity: props.equipment.quantity,
            stockItem: stockItem
        }
        
    }
    
    onRemove = () => {
        this.props.removeEquipment(this.state.id);
    }
    onQuantityChange = () => {

    }

    onPublicNameChange = () => {
        console.log('change !');
    }

    onStockNameChange = () => {

    }
    
    render() {
        return (
            <TableRow key={`${this.state.id}-row`}>
                <TableCell><Checkbox /></TableCell>
                <TableCell key = {`${this.state.id}-cell-1`}>{this.state.quantity}</TableCell>
                <TableCell key = {`${this.state.id}-cell-2`}>{this.state.stockItem.publicName}</TableCell>
                <TableCell key = {`${this.state.id}-cell-3`}>{this.state.stockItem.stockName}</TableCell>
                <TableCell key = {`${this.state.id}-cell-4`}>
                    <IconButton aria-label="Delete" onClick = {this.onRemove}>
                    <DeleteIcon />
                    </IconButton>
                </TableCell>   
            </TableRow>
        );
    };
}

const mapStateToProps = (state, props) => {
    return {
        equipments: state.equipments,
        filters: state.filters,
        stocklist : state.stocklist
    };
} 

const mapDispatchToProps = (dispatch, props) => {
    return {
        removeEquipment: (id) => dispatch(removeEquipment(id)),
        editEquipment: () => dispatch(editEquipment()),
        addEquipment: () => dispatch(addEquipment())
    };
}

export default connect(mapStateToProps,mapDispatchToProps)(EquipmentListItem)

