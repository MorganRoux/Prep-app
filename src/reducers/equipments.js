
const equipmentReducerDefaultState = [{
    category: 'microphone',
    brand: 'shure',
    description:'micro cardioide',
    quantity: 1,
    publicName: 'sm57',
    stockName: 'sm57A',
    id: 1
}, {
    category: 'microphone',
    brand: 'shure',
    description:'micro grosse caisse',
    quantity: 1,
    publicName: 'Beta91',
    stockName: 'B91',
    id: 2
}, {
    category: 'microphone',
    brand: 'shure',
    description:'micro grosse caisse',
    quantity: 5,
    publicName: 'Beta52',
    stockName: 'B52',
    id: 3
}];

const equipmentsReducer = (state = equipmentReducerDefaultState, action) => {
    switch(action.type){

    case 'REMOVE_EQUIPMENT' :
        const newState = state.filter((equipment) => (action.id !== equipment.id) );
        return newState;
    
    case 'EDIT_EQUIPMENT' :
    case 'ADD_EQUIPMENT' :
    default:
        return state;
    }
}

export default equipmentsReducer