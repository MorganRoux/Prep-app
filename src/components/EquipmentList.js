import React from 'react';
import EquipmentListItem from './EquipmentListItem';
import { connect } from 'react-redux'
import EquipmentKit from './EquipmentKit';

export const EquipmentList = (props) => (
    <div>
        <h3>EquipmentList</h3>
        
        { props.equipments.map((equipment) => {
            
            return (equipment.category === 'kit') ? (
                <EquipmentKit />
            ):(
                <EquipmentListItem 
                key={equipment.id}
                equipment = {equipment}
                /> 
            ); 
        })}
    </div>
);

const mapStateToProps = (state) => {
    return {
        equipments: state.equipments
    }
};

export default connect(mapStateToProps)(EquipmentList)