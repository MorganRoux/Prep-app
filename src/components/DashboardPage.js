import React from 'react';
import EquipmentList from './EquipmentList';

const DashboardPage = () => {
    return (
        <div>
            <h1>Dashboard Page</h1>
            <EquipmentList />
        </div>
        
    );
};

export default DashboardPage;