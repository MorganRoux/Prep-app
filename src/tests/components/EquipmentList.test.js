import React from 'react';
import { shallow } from 'enzyme';
import { EquipmentList, SortableComponent, EquipmentListSortable } from '../../components/EquipmentList';
import equipments from '../fixtures/equipment';

test('should render EquipmentList', () => {
    const wrapper = shallow(<EquipmentList equipments = {equipments}/>);
    expect(wrapper).toMatchSnapshot();

});

test('should render SortableComponent', () => {
    const moveEquipment = jest.fn();
    const wrapper = shallow(
        <SortableComponent 
            equipments = {equipments}
            moveEquipment = {moveEquipment}
        />);
    expect(wrapper).toMatchSnapshot();
});

test('should handle moveEquipment', () => {
    const oldIndex = 4;
    const newIndex = 1;
    const moveEquipment = jest.fn();
    const wrapper = shallow(
        <SortableComponent 
            equipments = {equipments}
            moveEquipment = {moveEquipment}
        />);
    wrapper.find(EquipmentListSortable).simulate('sortEnd',{oldIndex,newIndex});
    expect(moveEquipment).toHaveBeenLastCalledWith(oldIndex, newIndex)
})