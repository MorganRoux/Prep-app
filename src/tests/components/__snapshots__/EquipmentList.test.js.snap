// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Equipment List 1`] = `
<WithStyles(Paper)>
  <h3>
    EquipmentList
  </h3>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          <WithStyles(Checkbox) />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          key="quantity"
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-end"
            title="Trier"
          >
            <WithStyles(TableSortLabel)
              active={false}
              direction="asc"
              onClick={[Function]}
            >
              Quantit√©
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          key="name"
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-end"
            title="Trier"
          >
            <WithStyles(TableSortLabel)
              active={false}
              direction="asc"
              onClick={[Function]}
            >
              Nom
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          key="description"
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-end"
            title="Trier"
          >
            <WithStyles(TableSortLabel)
              active={false}
              direction="asc"
              onClick={[Function]}
            >
              Description
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)
      key="OJZIJDOIJAOD-body"
    >
      <Connect(Component)
        equipment={
          Object {
            "category": "microphone",
            "id": "OJZIJDOIJAOD",
            "parentId": null,
            "parentName": null,
            "quantity": 1,
            "stockName": "sm57",
          }
        }
        key="OJZIJDOIJAOD"
      />
    </WithStyles(TableBody)>
    <WithStyles(TableBody)
      key="OIJZUZKXKZ-body"
    >
      <Connect(Component)
        equipment={
          Object {
            "category": "microphone",
            "id": "OIJZUZKXKZ",
            "parentId": null,
            "parentName": null,
            "quantity": 1,
            "stockName": "B91",
          }
        }
        key="OIJZUZKXKZ"
      />
    </WithStyles(TableBody)>
    <WithStyles(TableBody)
      key="POIUYNBVBCFD-body"
    >
      <Connect(Component)
        equipment={
          Object {
            "category": "microphone",
            "id": "POIUYNBVBCFD",
            "parentId": null,
            "parentName": null,
            "quantity": 5,
            "stockName": "B52",
          }
        }
        key="POIUYNBVBCFD"
      />
    </WithStyles(TableBody)>
    <Connect(Component)
      key="PIUTRELJJGFHD"
      kit={
        Object {
          "category": "kit",
          "id": "PIUTRELJJGFHD",
          "parentId": null,
          "parentName": null,
          "quantity": 1,
          "stockName": "kit1",
        }
      }
    >
      <Connect(Component)
        equipment={
          Object {
            "category": "microphone",
            "id": "JHDJZJNZOCN",
            "parentId": "PIUTRELJJGFHD",
            "parentName": "kit1",
            "quantity": 3,
            "stockName": "sm57",
          }
        }
        key="JHDJZJNZOCN"
      />
      <Connect(Component)
        equipment={
          Object {
            "category": "microphone",
            "id": "UATEREGEBDNK",
            "parentId": "PIUTRELJJGFHD",
            "parentName": "kit1",
            "quantity": 10,
            "stockName": "B52",
          }
        }
        key="UATEREGEBDNK"
      />
      <Connect(Component)
        equipment={
          Object {
            "category": "microphone",
            "id": "POIAYLKJSDB",
            "parentId": "PIUTRELJJGFHD",
            "parentName": "kit1",
            "quantity": 5,
            "stockName": "B91",
          }
        }
        key="POIAYLKJSDB"
      />
    </Connect(Component)>
    <Connect(Component) />
  </WithStyles(Table)>
</WithStyles(Paper)>
`;
